### ISSUE TITLE
memory-store example 404s on callback due to trailing slash

### ISSUE THREAD

**Issue by bitemyapp** [StefanTerdell/userp#1]
![image](https://github.com/user-attachments/assets/5a8e1f31-543f-4931-b982-82bd17df2026)

Got a 404, "No webpage was found for the web address: `http://localhost:3000/signup/oauth/github/?code=blahdeeblahdeeblah`"

Looks like this route from `routes.rs` should be matching: 
https://github.com/StefanTerdell/userp/blob/main/src/routes.rs#L51

```rust
            signup_oauth_provider: "/signup/oauth/:provider",
```

It worked when I deleted the trailing slash: `http://localhost:3000/signup/oauth/github?code=blahdeeblahdeeblah`

I think I'd rather it be tolerant of trailing slashes? What do you think?

The dashboard after I fixed the URL, token redacted:

![image](https://github.com/user-attachments/assets/61f83af9-f6fd-4b4f-a05f-f08214e5e806)

**Comment by StefanTerdell** [comment]
This again! *shakes fist in random direction*

Thanks for opening an issue! I agree that this should be handled, but I'm not exactly sure how I'd like it to be. I'll elaborate in the PR.

**Comment by StefanTerdell** [comment]
Manually added trailing slashes to oauth callback routes fixed the issue for me, as discussed in #2! Closing for now, thanks :)
